### PR TITLE
(feat) O3-5423: Add unified REST error parsing utility

### PR DESCRIPTION
**Pull request by himanshumoral**
## Summary

Implements unified REST exception parsing utility for O3-5423 in `esm-error-handling`.

Normalizes different OpenMRS REST error shapes into a consistent `string[]` format for UI display.

Includes comprehensive unit tests.

## Related Issue

https://issues.openmrs.org/browse/O3-5423

### CODE REVIEW

**Comment by himanshumoral** [comment]
@jayasanka-sack @denniskigen @VeronicaMuthee Could you please review this PR?

File: packages/framework/esm-error-handling/src/index.ts [hunk 1]
```diff
@@ -20,60 +20,172 @@ window.onunhandledrejection = function (event: PromiseRejectionEvent) {
...
+ * @category Error Handling
  */
-export function createErrorHandler() {
-  const outgoingErr = Error();
-  return (incomingErr: unknown) => {
-    const finalErr = ensureErrorObject(incomingErr);
-    finalErr.stack += `\nAsync stacktrace:\n${outgoingErr.stack}`;
-    reportError(incomingErr);
-  };
+export function extractErrorMessagesFromResponse(error: unknown): Array<string> {
```
Comment (RajPrakash681) [inline packages/framework/esm-error-handling/src/index.ts]: refactor this as per reported by sonar

Comment (himanshumoral) [reply packages/framework/esm-error-handling/src/index.ts]: Refactored the function to reduce cognitive complexity and addressed the Sonar warning. Please let me know if any further changes are needed.

File: packages/framework/esm-error-handling/src/index.ts [hunk 2]
```diff
@@ -76,6 +76,145 @@ export function createErrorHandler() {
...
+
+/**
+ * The error body shape returned by the OpenMRS REST API.
+ *
+ * ValidationExceptions contain {@link fieldErrors} and/or {@link globalErrors}.
+ * Other exceptions contain {@link message} (and optionally {@link translatedMessage}).
+ *
+ * @category Error Handling
+ */
+export interface RestErrorResponse {
```
Comment (suubi-joshua) [inline packages/framework/esm-error-handling/src/index.ts]: Reviewing the existing logic/pattern 
`REST error → caught in ESM → reportError() → window.onerror → dispatchToastShown()`

I feel your new function sits completely outside this chain. It returns `string[]` for the caller to handle manually, meaning it doesn't integrate with `reportError`, `createErrorHandler`, or the global toast system at all. I think the issue at hand is to have a unified way to handle REST exceptions  but your changes seem to be adding a parallel system instead of extending the existing one.

**Comment by sohamdhande** [comment]
hey @himanshumoral 
The getRestErrorBody function checks 'error' in body but then immediately casts to RestErrorResponse via as unknown as RestErrorResponse — this bypasses the type system entirely and could silently pass a body that has an error key of the wrong shape (e.g. a string). A proper type guard with field-level narrowing would make this safe. Also, the test for empty fieldErrors/globalErrors asserts only expect.any(String) for subtitle, which doesn't actually verify which fallback branch fired — worth tightening to the expected ensureErrorObject message. Finally, reportRestError is a new public export and needs a changeset entry, otherwise it won't be included in any package version bump.
